lms/djangoapps/courseware/courses.py

```diff
@@ -141,6 +151,11 @@ def check_course_access(course, user, action, check_if_enrolled=False):
         if not CourseEnrollment.is_enrolled(user, course.id):
             raise CourseAccessRedirect(reverse('about_course', args=[unicode(course.id)]))
 
+    # Redirect if the user must answer a survey before entering the course.
+    if check_survey_complete:
+        if must_answer_survey(user, course):
```

Comment:
One last try, since I still don't love this name, but I didn't see a response to my last comment about this.  

Some alternative names:
- `is_survey_required_and_unanswered`
- `still_must_answer_survey`

The reason I don't love the name is because it is ambiguous and makes me think this is only checking `is_survey_required` (without necessarily the unanswered part).